src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpSyntaxNode.cs

```diff
@@ -21,22 +21,67 @@ internal abstract class HttpSyntaxNode : HttpSyntaxNodeOrToken
     {
     }
 
-    public override TextSpan Span => _span;
+    public override TextSpan FullSpan => _fullSpan;
+
+    public override bool IsSignificant
+    {
+        get
+        {
+            foreach (var child in ChildNodesAndTokens)
+            {
+                if (child.IsSignificant)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+    }
+
+    public override string Text => SourceText.ToString(Span);
+
+    public override TextSpan Span
+    {
+        get
+        {
+            var firstSignificantNodeOrToken = ChildNodesAndTokens
+                .FirstOrDefault(n => n.IsSignificant);
+
+            var lastSignificantNodeOrToken = ChildNodesAndTokens
+                .LastOrDefault(n => n.IsSignificant);
+
+            var startOfNonTriviaAndComments =
```

Comment:
I think something like `startOfSignificantText` may be less confusing - since comment and trivia tokens in the middle (after the first significant token and before the last significant token) are still considered part of the significant text...